includes/editor-templates/panel.php

```diff
@@ -189,12 +189,20 @@
 	<?php if ( ! Utils::has_pro() ) : ?>
 	<# if ( elementData.upsale_data && elementData.upsale_data.description ) { #>
 	<div class="elementor-nerd-box elementor-nerd-box--upsale">
+		<# if ( elementData.upsale_data.image ) { #>
+		<img class="elementor-nerd-box-icon" src="<?php Utils::print_unescaped_internal_string( esc_url( ELEMENTOR_ASSETS_URL . '{{ elementData.upsale_data.image }}' ) ); ?>" loading="lazy" alt="<?php echo esc_attr__( 'Upgrade', 'elementor' ); ?>" />
+		<# } else { #>
 		<img class="elementor-nerd-box-icon" src="<?php Utils::print_unescaped_internal_string( ELEMENTOR_ASSETS_URL . 'images/go-pro.svg' ); ?>" loading="lazy" alt="<?php echo esc_attr__( 'Upgrade', 'elementor' ); ?>" />
+		<# } #>
 		<# if ( elementData.upsale_data.title ) { #>
 		<div class="elementor-nerd-box-title">{{ elementData.upsale_data.title }}</div>
 		<# } #>
 		<div class="elementor-nerd-box-message">{{{ elementData.upsale_data.description }}}</div>
-		<a class="elementor-button go-pro" href="{{ elementData.upsale_data.upgrade_url }}" target="_blank"><?php echo esc_html__( 'Upgrade Now', 'elementor' ); ?></a>
+		<# if ( elementData.upsale_data.upgrade_text ) { #>
+		<a class="elementor-button go-pro" href='{{ elementData.upsale_data.upgrade_url }}' target="_blank"><?php echo esc_html( '{{ elementData.upsale_data.upgrade_text }}', 'elementor' ); ?></a>
```

Comment:
maybe move the if one level down ?
when you use the elementData.upsale_data.upgrade_text

Reply:
also . maybe already have default values in the array before sending the filter ?